core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -94,9 +105,15 @@ static TableMetadata newTableMetadata(Schema schema,
     }
     PartitionSpec freshSpec = specBuilder.build();
 
+    // TODO: case sensitivity
```

Comment:
This problem is not specific to `SortOrder` but also applies to `PartitionSpec`.

Reply:
I don't think this would be a problem unless the schema passed in is not the schema used to create the spec or sort order. If it is the same, then the case sensitive check will always work. If not, then we could have an error when we can't find the name when rebuilding for the schema with fresh IDs.

Reply:
Agree. We could have this problem but on one layer above.